helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithmFactory.java

```diff
@@ -61,19 +66,33 @@ public static RebalanceAlgorithm getInstance(
             new ReplicaActivateConstraint(), new NodeMaxPartitionLimitConstraint(),
             new ValidGroupTagConstraint(), new SamePartitionOnInstanceConstraint());
 
-    int evennessPreference =
-        preferences.getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS, 1);
-    int movementPreference =
-        preferences.getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.LESS_MOVEMENT, 1);
+    int evennessPreference = preferences
+        .getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS,
+            ClusterConfig.DEFAULT_GLOBAL_REBALANCE_PREFERENCE
+                .get(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS));
+    int movementPreference = preferences
+        .getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.LESS_MOVEMENT,
+            ClusterConfig.DEFAULT_GLOBAL_REBALANCE_PREFERENCE
+                .get(ClusterConfig.GlobalRebalancePreferenceKey.LESS_MOVEMENT));
+
+    boolean forceBaselineConverge = preferences
+        .getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.FORCE_BASELINE_CONVERGE, 0)
+        > 0;
+    if (forceBaselineConverge) {
+      MODEL.put(BaselineInfluenceConstraint.class.getSimpleName(), FORCE_BASELINE_CONVERGE_WEIGHT);
```

Comment:
I still suggest moving this after the soft constraint weight adjustment, and directly set to softConstraintsWithWeight instead of modifying the MODEL.

1. The 3 preference configurations shall be clearly defined. In theory, FORCE_BASELINE_CONVERGE
 is not related to EVENNESS or LESS_MOVEMENT. So we shall not let its static weight being polluted by the EVENNESS preference.
2. MODEL is a static map, but FORCE_BASELINE_CONVERGE can be changed dynamically. You cannot just enable it, but you will need to check and remove it if FORCE_BASELINE_CONVERGE is not setup. This is low efficiency. Better to directly put it into the dynamic map softConstraintsWithWeight. This is computed from scratch every time anyway.

Reply:
Makes sense, changing.